routes/users.js

```diff
@@ -405,7 +396,7 @@ router.post('/add_to_cart', function(req, res, next) {
         var cart;
 
         cart = user.cart;
-        if (cart === null | cart === undefined | Array.isArray(cart) & cart.length == 0) {
+        if (cart === null | cart === undefined | cart == []) {
```

Comment:
Doing `cart == null` checks for both being null and undefined. [Stack overflow](https://stackoverflow.com/a/2647888/6126802)